src/overlays/actors/ovl_En_Hintnuts/z_en_hintnuts.c

```diff
@@ -220,38 +220,27 @@ void EnHintnuts_SetupFreeze(EnHintnuts* this) {
 
 void EnHintnuts_Wait(EnHintnuts* this, GlobalContext* globalCtx) {
     s32 hasSlowPlaybackSpeed;
-    f32 boundedCurrentFrame;
-    f32 boundedCurrentFrameTemp;
 
     hasSlowPlaybackSpeed = 0;
     if (this->skelAnime.playSpeed < 0.5f) {
         hasSlowPlaybackSpeed = 1;
     }
-    if (hasSlowPlaybackSpeed && this->animFlagAndTimer != 0) {
-        this->animFlagAndTimer--;
+    if (hasSlowPlaybackSpeed) {
+        DECR(this->animFlagAndTimer);
     }
     if (Animation_OnFrame(&this->skelAnime, 9.0f) != 0) {
         this->collider.base.acFlags |= 1;
     } else if (Animation_OnFrame(&this->skelAnime, 8.0f) != 0) {
         Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_UP);
     }
-    if (this->skelAnime.curFrame < 9.0f) {
-        boundedCurrentFrame = 9.0f;
-    } else {
-        if (this->skelAnime.curFrame > 12.0f) {
-            boundedCurrentFrameTemp = 12.0f;
-        } else {
-            boundedCurrentFrameTemp = this->skelAnime.curFrame;
-        }
-        boundedCurrentFrame = boundedCurrentFrameTemp;
-    }
-    this->collider.dim.height = (((boundedCurrentFrame - 9.0f) * 9.0f) + 5.0f);
+
+    this->collider.dim.height = (((CLAMP(this->skelAnime.curFrame, 9.0f, 12.0f) - 9.0f) * 9.0f) + 5.0f);
```

Comment:
```suggestion
    this->collider.dim.height = 5.0f + ((CLAMP(this->skelAnime.curFrame, 9.0f, 12.0f) - 9.0f) * 9.0f);
```
Might be more clear this way, assuming it matches. Either way drop the outermost `()`.